pyresample/utils.py

```diff
@@ -692,3 +711,411 @@ def convert_def_to_yaml(def_area_file, yaml_area_file):
+    description : str, optional
+        Description/name of area. Defaults to area_id
+    proj_id : str, optional
+        ID of projection (being deprecated)
+    units : str, optional
+        Default projection units: meters, radians, or degrees
+    shape : list, optional
+        Number of pixels in the x and y direction (height, width)
+    area_extent : list, optional
+        Area extent as a list (lower_left_x, lower_left_y, upper_right_x, upper_right_y)
+    upper_left_extent : list, optional
+        Upper left corner of upper left pixel (x, y)
+    center : list, optional
+        Center of projection (x, y)
+    resolution : list or float, optional
+        Size of pixels: (dx, dy)
+    radius : list or float, optional
+        Length from the center to the edges of the projection (dx, dy)
+    rotation: float, optional
+        rotation in degrees or radians (negative is cw)
+    nprocs : int, optional
+        Number of processor cores to be used
+    lons : numpy array, optional
+        Grid lons
+    lats : numpy array, optional
+        Grid lats
+    optimize_projection:
+        Whether the projection parameters have to be optimized for a DynamicAreaDefinition.
+
+    Returns
+    -------
+    AreaDefinition or DynamicAreaDefinition : AreaDefinition or DynamicAreaDefinition
+        If shape and area_extent are found, an AreaDefinition object is returned.
+        If only shape or area_extent can be found, a DynamicAreaDefinition object is returned
+
+    Raises
+    ------
+    ValueError:
+        If neither shape nor area_extent could be found
+
+    Notes
+    -----
+    * **units** accepts 'deg', 'degrees', 'rad', 'radians', 'meters', and any parameter from cs2cs
+      (https://proj4.org/apps/cs2cs.html#cmdoption-cs2cs-lu). The order of default is:
+
+        1. units expressed with each variable through a DataArray's attrs attribute.
+        2. units passed to **units**
+        3. units used in **projection**
+        4. meters
+    * **resolution** and **radius** can be specified with one value if dx == dy
+    * If **resolution** and **radius** are provided as angles, center must be given or findable. In such a case,
+      they represent [projection x distance from center[0] to center[0]+dx, projection y distance from center[1] to
+      center[1]+dy]
+    """
+    from pyproj import Proj
+    description = kwargs.pop('description', area_id)
+    proj_id = kwargs.pop('proj_id', None)
+
+    # Get a proj4_dict from either a proj4_dict or a proj4_string.
+    proj_dict = _get_proj_data(projection)
+    p = Proj(proj_dict, preserve_units=True)
+
+    # If no units are provided, try to get units used in proj_dict. If still none are provided, use meters.
+    if units is None:
+        units = proj_dict.get('units', 'm')
+
+    # Makes sure list-like objects are list-like, have the right shape, and contain only numbers.
+    center = _verify_list('center', center, 2)
+    radius = _verify_list('radius', radius, 2)
```

Comment:
aren't these 2 allowed to have just one value ?

Reply:
Correct, radius and resolution may be just one value. The reason why I check for a size of 2 is because if a single number is provided, it is converted into a list of size 2 (which is internally how these variables function); there is not a case where a single number is provided such that my code will complain about a user only providing one value. If the user provides a list with only one element (rather than a stand-alone number), I interpret that as a mistake and do not allow that.